src/spikeinterface/curation/auto_merge.py

```diff
@@ -61,193 +82,373 @@ def get_potential_auto_merge(
+
+        # STEP : potential auto merge by correlogram
+        elif step == "correlogram" in steps:
+            correlograms_ext = sorting_analyzer.get_extension("correlograms")
+            if correlograms_ext is not None:
+                correlograms, bins = correlograms_ext.get_data()
+            else:
+                correlograms, bins = compute_correlograms(sorting, window_ms=window_ms, bin_ms=bin_ms, method="numba")
+            mask = (bins[:-1] >= -censor_correlograms_ms) & (bins[:-1] < censor_correlograms_ms)
+            correlograms[:, :, mask] = 0
+            correlograms_smoothed = smooth_correlogram(correlograms, bins, sigma_smooth_ms=sigma_smooth_ms)
+            # find correlogram window for each units
+            win_sizes = np.zeros(n, dtype=int)
+            for unit_ind in range(n):
+                auto_corr = correlograms_smoothed[unit_ind, unit_ind, :]
+                thresh = np.max(auto_corr) * adaptative_window_thresh
+                win_size = get_unit_adaptive_window(auto_corr, thresh)
+                win_sizes[unit_ind] = win_size
+            correlogram_diff = compute_correlogram_diff(
+                sorting,
+                correlograms_smoothed,
+                win_sizes,
+                pair_mask=pair_mask,
+            )
+            # print(correlogram_diff)
+            pair_mask = pair_mask & (correlogram_diff < corr_diff_thresh)
+            outs["correlograms"] = correlograms
+            outs["bins"] = bins
+            outs["correlograms_smoothed"] = correlograms_smoothed
+            outs["correlogram_diff"] = correlogram_diff
+            outs["win_sizes"] = win_sizes
+
+        # STEP : check if potential merge with CC also have template similarity
+        elif step == "template_similarity" in steps:
+            template_similarity_ext = sorting_analyzer.get_extension("template_similarity")
+            if template_similarity_ext is not None:
+                templates_similarity = template_similarity_ext.get_data()
+                templates_diff = 1 - templates_similarity
 
-        pair_mask = pair_mask & (templates_diff < template_diff_thresh)
-
-    # STEP 6 : validate the potential merges with CC increase the contamination quality metrics
-    if "check_increase_score" in steps:
-        pair_mask, pairs_decreased_score = check_improve_contaminations_score(
-            sorting_analyzer,
-            pair_mask,
-            contaminations,
-            firing_contamination_balance,
-            refractory_period_ms,
-            censored_period_ms,
-        )
+            else:
+                templates_ext = sorting_analyzer.get_extension("templates")
+                assert (
+                    templates_ext is not None
+                ), "auto_merge with template_similarity requires a SortingAnalyzer with extension templates"
+                templates_array = templates_ext.get_data(outputs="numpy")
+
+                templates_diff = compute_templates_diff(
```

Comment:
I know we want to keep the default behavior, but we want to quickly enforce the computation of template_similarity widget, since it will be more accurate than the one performed with lussac (because of the adaptative masks). Maybe for the refactoring of the fonction, we should list, for every preset, the requiered extensions and compute them if not present in the analyzer

Reply:
done